src/os.rs

```diff
@@ -576,27 +576,60 @@ mod imp {
 
 #[cfg(all(target_arch = "wasm32", not(target_os = "emscripten")))]
 mod imp {
-    use std::io;
+    use stdweb::unstable::TryInto;
+    use stdweb::web::error::Error as WebError;
+    use {Error, ErrorKind};
 
     #[derive(Debug)]
     pub struct OsRng;
 
     impl OsRng {
         pub fn new() -> Result<OsRng, Error> {
-            Err(Error::new(ErrorKind::Unavailable, "not supported on WASM"))
+            Ok(OsRng)
         }
+
         pub fn try_fill_bytes(&mut self, v: &mut [u8]) -> Result<(), Error> {
-            Err(Error::new(ErrorKind::Unavailable, "not supported on WASM"))
+            let len = v.len() as u32;
+            let ptr = v.as_mut_ptr() as i32;
```

Comment:
A 32-bit pointer? Surely that's not going to work on 64-bit platforms?

Reply:
Yes, it is `wasm32`.

Reply:
Okay, then could you at least `assert_eq!(size_of::<usize>(), 4)`?